src/main/java/uk/gov/justice/probation/courtcaseservice/service/ImmutableCourtCaseService.java

```diff
@@ -37,53 +38,55 @@
     private final CourtCaseRepository courtCaseRepository;
     private final TelemetryService telemetryService;
     private final GroupedOffenderMatchRepository matchRepository;
-    private final boolean globalProbationStatusUpdate;
+    private final OffenderRepository offenderRepository;
 
     @Autowired
     public ImmutableCourtCaseService(CourtRepository courtRepository,
                                      CourtCaseRepository courtCaseRepository,
                                      TelemetryService telemetryService,
                                      GroupedOffenderMatchRepository matchRepository,
-                                     @Value("${feature.flags.global-probation-status-update:true}") boolean globalProbationStatusUpdate) {
+                                     OffenderRepository offenderRepository) {
         this.courtRepository = courtRepository;
         this.courtCaseRepository = courtCaseRepository;
         this.telemetryService = telemetryService;
         this.matchRepository = matchRepository;
-        this.globalProbationStatusUpdate = globalProbationStatusUpdate;
+        this.offenderRepository = offenderRepository;
     }
 
     @Override
     public Mono<CourtCaseEntity> createCase(String caseId, CourtCaseEntity updatedCase) throws EntityNotFoundException, InputMismatchException {
         validateEntity(caseId, updatedCase);
+
+        updateOffenders(updatedCase, defendantEntity -> true);
         courtCaseRepository.findFirstByCaseIdOrderByIdDesc(caseId)
                 .ifPresentOrElse(
-                        (existingCase) -> {
+                        existingCase -> {
                             updatedCase.getDefendants()
-                                    .forEach(defendantEntity ->
-                                            updateOffenderMatches(existingCase, updatedCase, defendantEntity.getDefendantId())
-                                    );
-                            trackUpdateEvents(updatedCase, existingCase);
+                                    .forEach(defendantEntity -> updateOffenderMatches(existingCase, updatedCase, defendantEntity.getDefendantId()));
+                            trackUpdateEvents(existingCase, updatedCase);
```

Comment:
Yes - I switched the order of parameters. I noticed that confirmAndRejectMatches took existingCase first and updatedCase as second parameter and decided to make these other methods do the same.